@orta: Fixes for Kiosk CC testing on Staging, Unresolved bids, Bidding errors

ashfurrow opened the pull request:
Fixes:

- Testing Kiosks on staging
- Unresolved bids messaging
- [API errors resulting in being outbid](https://trello.com/c/tUeKka0c/70-kiosk-should-not-show-bidding-error-for-successful-bids-or-outbids)

ashfurrow commented:
- [ ] Add Stripe's test publishable key.
- [ ] Add CardFlight's API key and merchant account.
- [ ] Write code to use one or the other against the API depending on app settings.
- [ ] Test against staging with Stripe's test CC's.

orta commented on Kiosk/Bid Fulfillment/BidderNetworkModel.swift:
```diff
@@ -36,8 +36,12 @@ class BidderNetworkModel: NSObject {
+        // If the user exists, update their info to the API, otherwise create a new user.
         let signal = RACSignal.`if`(boolSignal, then: self.updateUser(), `else`: self.createNewUser())
+
+        // After update/create signal finishes, add a CC to their account (if we've collceted one)
```
typo: collected

orta commented on Kiosk/Bid Fulfillment/BidderNetworkModel.swift:
```diff
@@ -66,19 +70,22 @@ class BidderNetworkModel: NSObject {
         return updateProviderIfNecessary().then {
             self.fulfillmentController.loggedInProvider!.request(endpoint).filterSuccessfulStatusCodes().mapJSON()
-        }.doError { (error) in
+        }.logNext().doError { (error) in
```
nice

orta commented:
structurally it looks good, but there's a lot of code in here with changes and no tests

ashfurrow commented:
Cool, I'll add some tests.

ashfurrow commented on Kiosk/App/Views/SimulatorOnlyView.swift:
```diff
@@ -3,8 +3,13 @@ import UIKit
     override func awakeFromNib() {
-        let shouldShow = AppSetup.sharedState.showDebugButtons && AppSetup.sharedState.useStaging
-        let notTests = NSClassFromString("XCTest") == nil
-        self.hidden = !shouldShow && notTests
```
The logic here was messy, so I made it prettier.

ashfurrow commented:
OK, got the tests passing again, and added some minimal tests. `ConfirmYourBidPINViewController` needs to be split into a view model, I've opened https://github.com/artsy/eidolon/issues/511 to do so when I get a chance. 

The dev-only views were showing in some tests, which has been correct. Also, I changed the placeholder for the manual CC entry's year to be in the future.

orta commented on KioskTests/Bid Fulfillment/PlaceBidNetworkModelTests.swift:
```diff
@@ -0,0 +1,109 @@
+            expect(subject.bidderPosition?.id) == "5437dd107261692daa170000"
+        }
+
+        it("maps bid deatils into a proper request") {
```
chore: typo

orta commented on Podfile:
```diff
@@ -10,9 +10,12 @@ plugin 'cocoapods-keys', {
+    'StripeProductionPublishableKey',
+    'CardflightStagingAPIClientKey',
+    'CardflightStagingMerchantAccountToken',
+    'StripeStagingPublishableKey'
```
`make oss` will probably need an update too

ashfurrow replied:
We don't have a Makefile; both OSS users and CI use this lane, so we should be good.

orta commented:
Blocker, this test seems wrong, either in the logic or the name
<img width="894" alt="screen shot 2015-09-24 at 07 13 09" src="https://cloud.githubusercontent.com/assets/49038/10066544/d8ae450e-628b-11e5-8e14-54799c539a64.png">

ashfurrow commented:
No, that's correct. The bid has been submitted, we just don't know whether or not the user is the highest bidder. "Bid Submitted" is the most neutral language we could find. The previous screenshot was incorrect, as well, since it indicates you need to increase your bid (the test setup was wrong).

ashfurrow commented:
I'll rename the test, remove "error" – good idea.

orta commented:
Ace :+1:
